server/verify_test.go

```diff
@@ -651,56 +646,65 @@ func TestVerifyFailWithCertsAndPubkey(t *testing.T) {
 		IntermediateCerts: GceEKIntermediates,
 		TrustedAKs:        []crypto.PublicKey{priv.Public()},
 	}
-	if _, err := VerifyAttestation(att, opts); err == nil {
+	if err := validateOpts(opts); err == nil {
 		t.Error("Verified attestation even with multiple trust methods")
 	}
 }
 
-func TestVerifyAttestationEmptyRootsIntermediates(t *testing.T) {
+func TestParseAndValidateAK(t *testing.T) {
 	attestBytes := test.COS85NoNonce
 	att := &attestpb.Attestation{}
 	if err := proto.Unmarshal(attestBytes, att); err != nil {
 		t.Fatalf("failed to unmarshal attestation: %v", err)
 	}
 
-	if _, err := VerifyAttestation(att, VerifyOpts{
-		TrustedRootCerts:  nil,
-		IntermediateCerts: nil,
-	}); err == nil {
-		t.Error("expected error when calling VerifyAttestation with empty roots and intermediates")
-	}
-
-	if _, err := VerifyAttestation(att, VerifyOpts{}); err == nil {
-		t.Error("expected error when calling VerifyAttestation with empty VerifyOpts")
-	}
-}
-
-func TestVerifyAttestationMissingRoots(t *testing.T) {
-	attestBytes := test.COS85NoNonce
-	att := &attestpb.Attestation{}
-	if err := proto.Unmarshal(attestBytes, att); err != nil {
-		t.Fatalf("failed to unmarshal attestation: %v", err)
-	}
-
-	if _, err := VerifyAttestation(att, VerifyOpts{
-		IntermediateCerts: GceEKIntermediates,
-	}); err == nil {
-		t.Error("expected error when calling VerifyAttestation with missing roots")
+	testCases := []struct {
+		name     string
+		opts     VerifyOpts
+		wantPass bool
+	}{
+		{
+			name: "success",
+			opts: VerifyOpts{
+				TrustedRootCerts:  GceEKRoots,
+				IntermediateCerts: GceEKIntermediates,
+			},
+			wantPass: true,
+		},
+		{
+			name: "failed with empty roots and intermediates",
+			opts: VerifyOpts{
+				TrustedRootCerts:  nil,
+				IntermediateCerts: nil,
+			},
+			wantPass: false,
+		},
+		{
+			name:     "failed with empty VerifyOpts",
+			opts:     VerifyOpts{},
+			wantPass: false,
+		},
+		{
+			name:     "failed with missing roots",
+			opts:     VerifyOpts{IntermediateCerts: GceEKIntermediates},
+			wantPass: false,
+		},
+		{
+			name:     "failed with missing intermediates",
+			opts:     VerifyOpts{TrustedRootCerts: GceEKRoots},
+			wantPass: false,
```

Comment:
Add a test case setting TrustedAKs in VerifyOpts but not Trusted/Intermediate certs

Reply:
Done